utils/src/main/kotlin/Utils.kt

```diff
@@ -567,3 +572,50 @@ fun Throwable.showStackTrace() {
     // https://discuss.kotlinlang.org/t/if-operator-in-function-expression/7227.
     if (printStackTrace) printStackTrace()
 }
+
+/**
+ * Return Closable which temporarily moves the given directories to a temporary directory on initialization. On close
+ * it moves them back to their original location overwriting any conflicting existing directory.
+ */
+fun stashDirectories(vararg directories: File) : Closeable {
+    val directorySet = HashSet(Arrays.asList(*directories))
+    return StashDirectory(directorySet)
+}
+
+private class StashDirectory(directories: Set<File>) : Closeable {
+    private val stashedDirectories = mutableMapOf<File, File>()
+
+    init {
+        directories.forEach() {
+            stashDir(it)
+        }
+    }
+
+    override fun close() {
+        HashMap(stashedDirectories).forEach() {
+            unstashDir(it.key)
+        }
+    }
+
+    private fun stashDir(originalDir: File) {
+        if (!originalDir.isDirectory) return
+
+        val tempDir = createTempDir("analyzer", ".tmp", originalDir.parentFile)
+        log.info { "Temporarily moving directory from '${originalDir.absolutePath}' to '${tempDir.absolutePath}'." }
+
+        Files.move(originalDir.toPath(), tempDir.toPath(), StandardCopyOption.ATOMIC_MOVE)
+        stashedDirectories[originalDir] = tempDir
+    }
+
+    private fun unstashDir(originalDir: File) {
+        originalDir.safeDeleteRecursively()
+
+        val tempDir = stashedDirectories[originalDir]
+        tempDir ?: return
```

Comment:
Maybe nicer instead of this:

        stashedDirectories[originalDir]?.let { tempDir ->
            log.info { "Moving back directory from: '${tempDir.absolutePath}' to '${originalDir.absolutePath}'." }

            Files.move(tempDir.toPath(), originalDir.toPath(), StandardCopyOption.ATOMIC_MOVE)
            stashedDirectories.remove(originalDir)
        }

Reply:
I'd prefer to address this on top in order not to do too many things in this change.

Reply:
Ok.